gpm_api/tests/io/test_checks.py

```diff
@@ -0,0 +1,478 @@
+    with pytest.raises(ValueError):
+        checks.check_date(None)
+
+    # Test a string raises exception
+    with pytest.raises(ValueError):
+        checks.check_date('2014-12-31')
+
+
+def test_check_start_end_time() -> None:
+    ''' Check start and end time are valid '''
+
+    # Test a string
+    res = checks.check_start_end_time('2014-12-31', '2015-01-01')
+    assert isinstance(res, tuple)
+
+    # Test the reverse for exception
+    with pytest.raises(ValueError):
+        checks.check_start_end_time('2015-01-01', '2014-12-31')
+
+    # Test a datetime object
+    res = checks.check_start_end_time(
+        datetime.datetime(2014, 12, 31), datetime.datetime(2015, 1, 1)
+    )
+    assert isinstance(res, tuple)
+
+    # Test the reverse datetime object for exception
+    with pytest.raises(ValueError):
+        checks.check_start_end_time(
+            datetime.datetime(2015, 1, 1),
+            datetime.datetime(2014, 12, 31)
+        )
+
+    # Test a datetime timestamp with h/m/s/ms
+    res = checks.check_start_end_time(
+        datetime.datetime(2014, 12, 31, 12, 30, 30, 300),
+        datetime.datetime(2015, 1, 1, 12, 30, 30, 300)
+    )
+
+    # Test end time in the future
+    with pytest.raises(ValueError):
+        checks.check_start_end_time(
+            datetime.datetime(2014, 12, 31, 12, 30, 30, 300),
+            datetime.datetime(2125, 1, 1, 12, 30, 30, 300)
+        )
+
+    # Test start time in the future
+    with pytest.raises(ValueError):
+        checks.check_start_end_time(
+            datetime.datetime(2125, 12, 31, 12, 30, 30, 300),
+            datetime.datetime(2126, 1, 1, 12, 30, 30, 300)
+        )
+
+
+def test_check_scan_mode(
+    versions: list[int],
+    products: list[str],
+) -> None:
+    ''' Check scan mode is valid '''
+
+    for product in products:
+        for version in versions:
+            # Get available scan modes
+            scan_modes = available_scan_modes(product, version)
+
+            for scan_mode in scan_modes:
+                res = checks.check_scan_mode(scan_mode, product, version)
+                assert res == res, (
+                    f"Function returned {res} for scan_mode {scan_mode}, "
+                    f"expected {scan_mode}"
+                )
+
+            # Test a scan mode that doesn't exist
+            for scan_mode in ['NOT', 'A', 'SCAN', 'MODE']:
+                with pytest.raises(ValueError):
+                    checks.check_scan_mode(scan_mode, product, version)
+
+            # Try to have function infer scan mode
+            res = checks.check_scan_mode(None, product, version)
+            assert res in scan_modes, (
+                f"Function returned {res} for scan_mode {scan_mode}, "
+                f"expected {scan_mode}"
+            )
+
+            # Test a scan mode that isn't a string
+            with pytest.raises(ValueError):
+                checks.check_scan_mode(123, product, version)
+
+
+def test_check_bbox() -> None:
```

Comment:
We could remove this test and the `check_bbox` function since not used anymore across the code. Sorry ... I overlooked that

Reply:
Maybe the test code could be used to test something related to the `extent`  in the `utils.geospatial.py` module